internal/spdxlicense/generate/generate_license_list.go

```diff
@@ -139,9 +175,26 @@ func processSPDXLicense(result LicenseList) map[string]string {
 		}
 	}
 
+	for k, v := range licenseIDs {
+		cleanID := strings.ToLower(v)
```

Comment:
Why do we recalculate the `cleanID`? I thought the `k` value was the clean ID, based on looking above at line 151

Reply:
Because `v` is used as "primary key", that is: if `k0`, `k1`, `kn` point to v0, and in the overwrite map `v0` should point to `v1`, than the loop aboce will apply that to `k0`, `k1`, `kn`.

Reply:
I think dan is right in the sense that `k` is no different than `cleanID`; take:
https://github.com/anchore/syft/pull/1009/files#diff-122a77148a7f1f5f9c1e3e3715c303dd5ce8d0ba75d9348ec00a1bfd0a46c305L117-L122
```golang
	for _, l := range result.Licenses {
		cleanID := strings.ToLower(l.ID)
		...
		licenseIDs[cleanID] = l.ID
                ...
         }
```

So later when doing:
```
for k, v := range licenseIDs {
   ...
}
```
`k` is indeed `cleanID`. This could be written like so for more clarity too:
```
for cleanID, id := range licenseIDs {
   ...
}
```

Shout out if I'm missing something

Reply:
Alright this can be rewritten as 
```go
	for cleanID, realID := range licenseIDs {
		if replacementID, exists := overwrite[strings.ToLower(realID)]; exists {
			licenseIDs[cleanID] = replacementID
		}
	} 
```
This code is replacing `cleanIDs` based on what value they point to. so if permutation `abc-1.0.0.0+` points to `ABC-1.0+`, and the overwrite map says `ABC-1.0+` should point to `ABC-1.0-Or-Later` that will happen for `abc-1.0.0.0+`, but also for all of its permutations, since they all point to `ABC-1.0+`.

Mind that `ABC-1.0+` is stored in the overwrite map as `abc-1.0+` (lower case),  so the use of `strings.ToLower(realID)` to find it in the map.

I suppose the reuse of the cleanID name, as in sanitized id for map indexation, caused more confusion than clarification.